src/baseline_system/parsers/props_wrapper.py

```diff
@@ -514,22 +515,40 @@ def _gensym_ent(self):
                       get_zero_args = False,
                       get_conj = False)
 
-    okrs = []
-    for line in open(input_fn):
-        sent = line.strip()
 
-        if not sent or sent.startswith('#'):
-            # Ignore commented lines
-            continue
+    if (input_fn is not None) and (output_fn is not None):
+        okrs = []
+        for line in open(input_fn):
+            sent = line.strip()
 
-        # 2. Parse sentence
-        pw.parse(sent)
+            if not sent or sent.startswith('#'):
+                # Ignore commented lines
+                continue
 
-        # 3. Get OKR Json object
-        okr = pw.get_okr()
-        okrs.append(okr)
-        logging.info(pformat(okr))
+            # 2. Parse sentence
+            pw.parse(sent)
+
+            # 3. Get OKR Json object
+            okr = pw.get_okr()
+            okrs.append(okr)
+            logging.info(pformat(okr))
+
+        # Dump json
+        with open(output_fn, 'w') as fout:
+            json.dump(okrs, fout)
+
+    else:
+        # Interactive shell
+        logging.info("Input a raw sentence. Press enter to finish each sentence.")
```

Comment:
why logging.info and not simply "print"?

Reply:
See my answer below.
Okay to merge, other than that?

Reply:
Sure. merged.